src/tools.cc

```diff
@@ -1110,6 +1110,7 @@ restoreCapabilities(bool keep)
         cap_list[ncaps] = CAP_NET_BIND_SERVICE;
         ++ncaps;
         if (Ip::Interceptor.TransparentActive() ||
+                Ip::Interceptor.InterceptActive() ||
```

Comment:
Why? the existing Netfilter operations apparently do not need this level of privileges, so there should be no reason to add them now.

Reply:
Without this capability netfilter says 'operation not permitted' when trying to get a connection mark. This makes sense, because we try to get a mark of a connection that is not created by squid. We query details of intercepted connection (that we do not own), so appropriate capability is a must here.

Reply:
I think you have found a Linux bug then. It should not require root level privileges for *reading* socket state with getsockopt(), only for changing the state with setsockopt(). Please add a TODO about looking into better restrictions later.